src/modules/auth/jwt.strategy.spec.ts

```diff
@@ -0,0 +1,117 @@
+import { Test, TestingModule } from '@nestjs/testing';
+import { createMock } from '@golevelup/ts-jest';
+import { UserUseCases } from '../user/user.usecase';
+import { JwtStrategy } from './jwt.strategy';
+import { newUser } from '../../../test/fixtures';
+import { ConfigService } from '@nestjs/config';
+import { UnauthorizedException } from '@nestjs/common';
+import { getTokenDefaultIat } from '../../lib/jwt';
+
+describe('Jwt strategy', () => {
+  let userUseCases: UserUseCases;
+  let strategy: JwtStrategy;
+
+  beforeEach(async () => {
+    const moduleRef = await createTestingModule();
+    userUseCases = moduleRef.get<UserUseCases>(UserUseCases);
+    strategy = moduleRef.get<JwtStrategy>(JwtStrategy);
+  });
+
+  it('When token is old, then fail', async () => {
+    await expect(strategy.validate({ email: 'test@test.com' })).rejects.toThrow(
+      new UnauthorizedException('Old token version detected'),
+    );
+  });
+
+  it('When user does not exist, then fail', async () => {
+    jest.spyOn(userUseCases, 'getUser').mockResolvedValue(null);
+
+    await expect(
+      strategy.validate({ payload: { uuid: 'anyUuid' } }),
+    ).rejects.toThrow(UnauthorizedException);
+  });
+
+  it('When token iat is older than lastPasswordChangedAt , then fail', async () => {
+    const user = newUser();
+    const greaterDate = new Date();
```

Comment:
To ensure that this case is not failing, please, do use the token iat as the base date for a new Date, and the add the minutes. If the tests start to run slowly, this test could not pass. The tests should be consistent always. For instance: 

```
const user = newUser();
const tokenIat = getTokenDefaultIat();
const olderDate = new Date(tokenIat);
olderDate.setMinutes(olderDate.getMinutes() + 1);
// and so on ..
```